### PR TITLE
[refactor] Improve utils.GetTypeName

### PR DESCRIPTION
**Pull request by lukesneeringer**
Having different methods is duplicative. One is better.

### CODE REVIEW

File: rules/internal/utils/type_name.go [hunk 1]
```diff
@@ -17,23 +17,17 @@ package utils
...
-	return strings.ToLower(t.GetType().String()[len("TYPE_"):])
-}
-
-// GetMessageTypeName returns the name of the type of the field, as a string.
-func GetMessageTypeName(f *desc.FieldDescriptor) string {
-	if messageType := f.GetMessageType(); messageType != nil {
-		return messageType.GetFullyQualifiedName()
+// GetTypeName returns the name of the type of the field, as a string,
+// regardless
+func GetTypeName(f *desc.FieldDescriptor) string {
```
Comment (mingzhi) [inline rules/internal/utils/type_name.go]: Add a support for map type too.

https://godoc.org/github.com/jhump/protoreflect/desc#FieldDescriptor.GetMapKeyType
https://godoc.org/github.com/jhump/protoreflect/desc#FieldDescriptor.GetMapValueType

File: rules/internal/utils/type_name.go [hunk 2]
```diff
@@ -17,23 +17,17 @@ package utils
...
+	if m := f.GetMessageType(); m != nil {
+		return m.GetFullyQualifiedName()
```
Comment (mingzhi) [inline rules/internal/utils/type_name.go]: Are you sure to return the fully qualified name?
For example:
```
package test;
message Book {
  Author author = 1;
}
message Author{}
```
Will it return "test.Author" if we get the type name of `author`? But we really want "Author", right?

File: rules/internal/utils/type_name_test.go [hunk 3]
```diff
@@ -20,43 +20,22 @@ import (
 	"github.com/googleapis/api-linter/rules/internal/testutils"
 )
 
-func TestGetScalarTypeName(t *testing.T) {
-	for _, ty := range []string{"int32", "int64", "string", "bytes", "float", "double"} {
+func TestGetTypeName(t *testing.T) {
+	for _, ty := range []string{"int32", "int64", "string", "bytes", "google.protobuf.Timestamp", "Format"} {
 		t.Run(ty, func(t *testing.T) {
 			file := testutils.ParseProto3Tmpl(t, `
+				import "google/protobuf/timestamp.proto";
```
Comment (mingzhi) [inline rules/internal/utils/type_name_test.go]: Add "package test.v1;".

File: rules/internal/utils/type_name.go (hunk 2 above)
Comment (lukesneeringer) [reply rules/internal/utils/type_name.go]: I always want the fully qualified name in this case.

File: rules/internal/utils/type_name.go (hunk 1 above)
Comment (lukesneeringer) [reply rules/internal/utils/type_name.go]: Reasonable, but can I do it in a follow-up?

Comment (mingzhi) [reply rules/internal/utils/type_name.go]: Sure. Can you add a TODO?

Comment (lukesneeringer) [reply rules/internal/utils/type_name.go]:
```suggestion
//
// TODO: Add support for map types.
func GetTypeName(f *desc.FieldDescriptor) string {
```

**Pull request merged**